### ISSUE TITLE
JavadocLeadingAsteriskAlign reports false positives

### ISSUE THREAD

**Issue by gianmarcoschifone** [checkstyle/checkstyle#20958]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadocleadingasteriskalign.html
I have downloaded the latest Checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the CLI and showed it below, as CLI describes the problem better than 1,000 words.

## Description

This was noticed from the PR discussion:

- https://github.com/checkstyle/checkstyle/pull/20934#issuecomment-5053294087
- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/13c94aa45ddc5c5a7884e5cd9acbf3f33416f5a1_2026205613/reports/diff/checkstyle/index.html#A7

Example:

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % cat Test.java
public interface Test {
    /**   *  ***
     * There is a space, then a leading asterisk, then a space,
     * then asterisks in the first line
     */
    void test();
}
```

Config:

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocLeadingAsteriskAlign"/>
  </module>
</module>
```

Actual result:

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % java -jar checkstyle-13.9.0-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
[ERROR] Test.java:3:6: Leading asterisk has incorrect indentation level 6, expected is 9. [JavadocLeadingAsteriskAlign]
[ERROR] Test.java:4:6: Leading asterisk has incorrect indentation level 6, expected is 9. [JavadocLeadingAsteriskAlign]
[ERROR] Test.java:5:6: Leading asterisk has incorrect indentation level 6, expected is 9. [JavadocLeadingAsteriskAlign]
Audit done.
Checkstyle ends with 3 errors.
```

Expected result:

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % java -jar checkstyle-13.9.0-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

## Javadoc JDK Tool

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % javadoc -d docs -private Test.java
Loading source file Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21.0.8+9-LTS
Building tree for all the packages and classes...
Generating docs/Test.html...
Test.java:1: warning: no comment
public interface Test {
       ^
Generating docs/package-summary.html...
Copying file ./doc-files/image.png to directory docs/doc-files...
Generating docs/package-tree.html...
Generating docs/overview-tree.html...
Building index for all classes...
Generating docs/allclasses-index.html...
Generating docs/allpackages-index.html...
Generating docs/index-all.html...
Generating docs/search.html...
Generating docs/index.html...
Generating docs/help-doc.html...
1 warning
```

<img width="1038" height="127" alt="Image" src="https://github.com/user-attachments/assets/130b693f-4423-4fc4-8c86-bf46871910cb" />

## AST

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % java -jar checkstyle-13.9.0-SNAPSHOT-all.jar -J Test.java
COMPILATION_UNIT -> COMPILATION_UNIT [1:1]
`--INTERFACE_DEF -> INTERFACE_DEF [1:1]
    |--MODIFIERS -> MODIFIERS [1:1]
    |   `--LITERAL_PUBLIC -> public [1:1]
    |--LITERAL_INTERFACE -> interface [1:8]
    |--IDENT -> Test [1:18]
    `--OBJBLOCK -> OBJBLOCK [1:23]
        |--LCURLY -> { [1:23]
        |--METHOD_DEF -> METHOD_DEF [6:5]
        |   |--MODIFIERS -> MODIFIERS [6:5]
        |   |--TYPE -> TYPE [6:5]
        |   |   |--BLOCK_COMMENT_BEGIN -> /* [2:5]
        |   |   |   |--COMMENT_CONTENT -> *   *  ***\n     * There is a space, then a leading asterisk, then a space,\n     * then asterisks in the first line\n      [2:7]
        |   |   |   |   `--JAVADOC_CONTENT -> JAVADOC_CONTENT [2:11]
        |   |   |   |       |--LEADING_ASTERISK -> * [2:11]
        |   |   |   |       |--TEXT ->   *** [2:12]
        |   |   |   |       |--NEWLINE -> \n [2:17]
        |   |   |   |       |--LEADING_ASTERISK -> * [3:6]
        |   |   |   |       |--TEXT ->  There is a space, then a leading asterisk, then a space, [3:7]
        |   |   |   |       |--NEWLINE -> \n [3:64]
        |   |   |   |       |--LEADING_ASTERISK -> * [4:6]
        |   |   |   |       |--TEXT ->  then asterisks in the first line [4:7]
        |   |   |   |       |--NEWLINE -> \n [4:40]
        |   |   |   |       `--TEXT ->       [5:1]
        |   |   |   `--BLOCK_COMMENT_END -> */ [5:6]
        |   |   `--LITERAL_VOID -> void [6:5]
        |   |--IDENT -> test [6:10]
        |   |--LPAREN -> ( [6:14]
        |   |--PARAMETERS -> PARAMETERS [6:15]
        |   |--RPAREN -> ) [6:15]
        |   `--SEMI -> ; [6:16]
        `--RCURLY -> } [7:1]
```

**Comment by romani** [comment]
@gianmarcoschifone , please share CLI output as we recommend in issue template.
Please also share screenshot of html rendering of such javadoc to be sure how javadoc interpret this

**Comment by gianmarcoschifone** [comment]
@romani done

**Comment by romani** [comment]
>Please also share screenshot of html rendering of such javadoc to be sure how javadoc interpret this

please do

**Comment by gianmarcoschifone** [comment]
> please do

@romani done, sorry

**Comment by romani** [comment]
@gianmarcoschifone , please share AST print by CLI for this Test.java 

sounds like this will be WONT fix.

**Comment by gianmarcoschifone** [comment]
> [@gianmarcoschifone](https://github.com/gianmarcoschifone) , please share AST print by CLI for this Test.java
> 
> sounds like this will be WONT fix.

@romani done in issue description

**Comment by romani** [comment]
violation in general is ok, BUT
technically we report wrong column
```
        |   |   |   |       |--LEADING_ASTERISK -> * [2:11]
        |   |   |   |       |--TEXT ->   *** [2:12]
        |   |   |   |       |--NEWLINE -> \n [2:17]
        |   |   |   |       |--LEADING_ASTERISK -> * [3:6]
```

as it should be `Leading asterisk has incorrect indentation level 6, expected is 11.` , **expected is 11**.

@gianmarcoschifone , do you agree ?

**Comment by gianmarcoschifone** [comment]
@romani Trying with JDK tool looks like it always recognizes the asterisk as part of javadoc syntax until the second group (where the first group is `/**` with many asterisks as desired).
For example also this one

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % cat Test.java
public interface Test {
    /***  **   *  *   *  *
     * There is a space, then a leading asterisk, then a space,
     * then asterisks in the first line
     */
    void test();
}
```

<img width="623" height="128" alt="Image" src="https://github.com/user-attachments/assets/2dc87192-0c97-49ee-9d89-f0119e47e596" />

So if we want to consider `leading asterisk` as the last one the JDK tool includes in its syntax (and so doesn't make it appear in html render), then yes probably column 11 for that input.

Other examples:

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % cat Test.java
public interface Test {
    /*****  *   *  *   *  *
     * There is a space, then a leading asterisk, then a space,
     * then asterisks in the first line
     */
    void test();
}
```

<img width="587" height="125" alt="Image" src="https://github.com/user-attachments/assets/2af4c316-1bfb-4464-a740-064796e03e71" />

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % cat Test.java
public interface Test {
    /**  *   *  *   *  *
     * There is a space, then a leading asterisk, then a space,
     * then asterisks in the first line
     */
    void test();
}
```

<img width="578" height="117" alt="Image" src="https://github.com/user-attachments/assets/631b9b63-68c9-40db-83cb-12ff5ee86ceb" />

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % cat Test.java
public interface Test {
    /**  *   *  *   *
     * There is a space, then a leading asterisk, then a space,
     * then asterisks in the first line
     */
    void test();
}
```

<img width="573" height="122" alt="Image" src="https://github.com/user-attachments/assets/c694fa4b-b672-4488-94ee-f5140bb75b7b" />

**Comment by romani** [comment]
lets not participate in this madness of leading asterisks, 

lets do simple match of columns of previous Leading asterisk. our parser made it at column 11, so be it.

@arturict , can you help us to fix it ? we need to be accurate and not be involved in big updates. As this edge case, nobody care on precise correctness on such messy javadoc. But if solution can be simple, we can fix.

**Comment by arturict** [comment]
Done in #20966. It keeps the violations and uses the opening-line `LEADING_ASTERISK` as the expected column only when it is separated from the opening marker by whitespace. The regression test verifies expected column 11 in all three messages.

**Comment by arturict** [comment]
Could you add the approved label? The PR description check requires it.

**Comment by romani** [comment]
fix is merged
